src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,589 @@
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=ctrl,
+            cluster=Clusters.IcdManagement,
+            attribute=Clusters.IcdManagement.Attributes.IdleModeDuration
+        )
+
+    def verify_attribute_data(self, sub, cluster, attribute, ep=0):
+        sub_attrs = sub
+        if isinstance(sub, Clusters.Attribute.SubscriptionTransaction):
+            sub_attrs = sub.GetAttributes()
+
+        asserts.assert_true(ep in sub_attrs, "Must have read endpoint %s data" % ep)
+        asserts.assert_true(cluster in sub_attrs[0], "Must have read %s cluster data" % cluster.__name__)
+        asserts.assert_true(attribute in sub_attrs[0][cluster],
+                            "Must have read back attribute %s" % attribute.__name__)
+
+    def get_attribute_from_sub_dict(self, sub, cluster, attribute, ep=0):
+        return sub[ep][cluster][attribute]
+
+    def get_typed_attribute_path(self, attribute, ep=0):
+        return TypedAttributePath(
+            Path=AttributePath(
+                EndpointId=ep,
+                Attribute=attribute
+            )
+        )
+
+    async def get_dut_acl(self):
+        sub = await self.default_controller.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=[(0, Clusters.AccessControl.Attributes.Acl)],
+            keepSubscriptions=False,
+            fabricFiltered=True
+        )
+
+        acl_list = self.get_attribute_from_sub_dict(
+            sub=sub,
+            cluster=Clusters.AccessControl,
+            attribute=Clusters.AccessControl.Attributes.Acl
+        )
+
+        return acl_list
+
+    def is_uint32(self, var):
+        return isinstance(var, int) and 0 <= var <= 4294967295
+
+    def is_uint16(self, var):
+        return isinstance(var, int) and 0 <= var <= 65535
+
+    @async_test_body
+    async def test_TC_IDM_4_2(self):
+
+        # Test setup
+        cluster_rev_attr = Clusters.BasicInformation.Attributes.ClusterRevision
+        cluster_rev_attr_typed_path = self.get_typed_attribute_path(cluster_rev_attr)
+        node_label_attr = Clusters.BasicInformation.Attributes.NodeLabel
+        node_label_attr_path = [(0, node_label_attr)]
+        node_label_attr_typed_path = self.get_typed_attribute_path(node_label_attr)
+        SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 0
+        INVALID_ACTION_ERROR_CODE = 0x580
+        same_min_max_interval_sec = 3
+
+        # Controller 1 setup
+        CR1: ChipDeviceController = self.default_controller
+
+        # Controller 2 setup
+        fabric_admin = self.certificate_authority_manager.activeCaList[0].adminList[0]
+        CR2_nodeid = self.matter_test_config.controller_node_id + 1
+        CR2: ChipDeviceController = fabric_admin.NewController(
+            nodeId=CR2_nodeid,
+            paaTrustStorePath=str(self.matter_test_config.paa_trust_store_path),
+        )
+
+        # Get DUT ACL
```

Comment:
this comment does not seem to help reading the code: I already see `... self.get_dut_acl` so this obviously gets the DUT ACL.

Is there something that can be described to help the reader of the code better understand things? if not, lets drop the comment. I think this applies to some more comments below as well.

Reply:
since ACL management is a bit part of this CL, I wonder if it would help if we would have a separate class only for ACL management. Something that has a `Admin Controller` and remembers the original ACL (just like this does) and then allows ACL altering. Could be a followup as well if the design of such a class is not immediately obvious (I am unclear about how to name methods on it for example)

Reply:
It sounds good as a follow up, so I can give it proper time to design.